### ISSUE TITLE
Newly added carriers don't appear in FO checkout - product v1

### ISSUE THREAD

**Issue by arouiadib** [PrestaShop/PrestaShop#24498]
#### Describe the bug
If you add a new active carrier in BO, and purchase a product, the carrier doesn't appear in FO checkout.

#### Expected behavior

If a carrier is added and is active, it should appear in checkout page.
(Well, there is an additional condition for a carrier to appear in checkout is to be selected in "available carriers" for the specific product, but this is a separate concern).

#### Steps to Reproduce

0. Have a fresh install on develop
1. Go to Shipping > Carriers, add a new carrier "Test carrier"
2. Go to FO > order a product (non virtual) > in checkout page see "Test carrier" is not on the list under Shipping tab.

#### Additional information

* PrestaShop version: develop
* PHP version: 7.3

**Comment by florine2623** [comment]
Hello @arouiadib ,

I can't reproduce your issue.
Please check my screen record : https://drive.google.com/file/d/1LBUw8OHpYVrXC_2kESgDphP4mDddr8GZ/view

I'll close your issue now. Feel free to open a new one when more details appear.

**Comment by arouiadib** [comment]
Thanks @florine2623 for the time creating the screen record.

I found the mistake I was doing: the customer address was in US while my newly created carrier doesn't contain US under "Carrier Edit page > Shipping Locations and costs". Shame on me :sweat: 

To understand more what are the conditions under which a carrier appears in FO checkout, and for the sake of history,  I have created an issue :https://github.com/PrestaShop/prestashop-specs/issues/220 , Your contribution will be appreciated.

Again thanks
